internal/auth/netauth/netauth.go

```diff
@@ -0,0 +1,132 @@
+package netauth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/foxcpp/maddy/framework/address"
+	"github.com/foxcpp/maddy/framework/config"
+	"github.com/foxcpp/maddy/framework/log"
+	"github.com/foxcpp/maddy/framework/module"
+	"github.com/hashicorp/go-hclog"
+	"github.com/netauth/netauth/pkg/netauth"
+)
+
+const modName = "auth.netauth"
+
+func init() {
+	var _ module.PlainAuth = &Auth{}
+	var _ module.Table = &Auth{}
+	module.Register(modName, New)
+	module.Register("table.netauth", New)
+}
+
+// Auth binds all methods related to the NetAuth client library.
+type Auth struct {
+	instName  string
+	mustGroup string
+
+	nacl *netauth.Client
+
+	log log.Logger
+}
+
+// New creates a new instance of the NetAuth module.
+func New(modName, instName string, _, inlineArgs []string) (module.Module, error) {
+	return &Auth{
+		instName: instName,
+		log:      log.Logger{Name: modName},
+	}, nil
+}
+
+// Init performs deferred initialization actions.
+func (a *Auth) Init(cfg *config.Map) error {
+	a.log.Println("In netauth init")
+	l := hclog.New(&hclog.LoggerOptions{Output: a.log})
+	n, err := netauth.NewWithLog(l)
+	if err != nil {
+		return err
+	}
+	a.nacl = n
+	a.nacl.SetServiceName("maddy")
+	cfg.String("require_group", false, false, "", &a.mustGroup)
+	cfg.Bool("debug", true, false, &a.log.Debug)
+	if _, err := cfg.Process(); err != nil {
+		return err
+	}
+
+	a.log.Debugln("Debug logging enabled")
+	a.log.Debugf("mustGroups status: %s", a.mustGroup)
+	return nil
+}
+
+// Name returns "auth.netauth" as the fixed module name.
+func (a *Auth) Name() string {
+	return modName
+}
+
+// InstanceName returns the configured name for this instance of the
+// plugin.  Given the way that NetAuth works it doesn't really make
+// sense to have more than one instance, but this is part of the API.
+func (a *Auth) InstanceName() string {
+	return a.instName
+}
+
+// Lookup requests the entity from the remote NetAuth server,
+// potentially returning that the user does not exist at all.
+func (a *Auth) Lookup(ctx context.Context, username string) (string, bool, error) {
+	localpart, _, err := address.Split(username)
```

Comment:
Probably needs to be removed, how such authentication providers are intended to be used is described in https://maddy.email/tutorials/pam/#account-names

Reply:
Changed per directions.  Those documents explain how to do this with the imap storage, but submission also references the auth block.  Can you add some to the docs about how to ensure that that also does the translation?

Reply:
storage.imapsql is used by both delivery (submission) and IMAP.  delivery_map is applied specifically for delivery part. 

The IMAP part is supposed to be handled by auth_map, but that's not necessary in PAM/netauth cases - one could just use usernames without domain part as a login.

Reply:
Right, and I think it should be because it is not clear in the tutorial when you should/should not log in as an unqualified local name.